lib/Sema/CSApply.cpp

```diff
@@ -4196,14 +4196,22 @@ namespace {
       // The declaration we found must be exposed to Objective-C.
       tc.validateDecl(method);
       if (!method->isObjC()) {
-        tc.diagnose(E->getLoc(), diag::expr_selector_not_objc,
-                    foundDecl->getDescriptiveKind(), foundDecl->getFullName())
+          // If the method declaration lies in a protocol and we're providing
+          // a default implementation of the method through a protocol extension
+          // then insert the fixit on protocol, rather than on the method in the
+          // protocol declaration (not allowed).
+          auto protocolDecl = foundDecl->getDeclContext()->getSelfProtocolDecl();
+          
+          tc.diagnose(E->getLoc(), diag::expr_selector_not_objc,
+                      foundDecl->getDescriptiveKind(), foundDecl->getFullName())
           .highlight(subExpr->getSourceRange());
```

Comment:
Nitpick: these chaining lines are supposed to be indented. (In our usual style continuation lines are double-indented, but we get sloppy about it for diagnostics sometimes because we chain so frequently, and in this case the dots are lining up.)

Reply:
Done! 👍